src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/InventoryServiceImplIT.java

```diff
@@ -0,0 +1,182 @@
+            .channels()
+            .post(channelDraft1)
+            .execute()
+            .toCompletableFuture()
+            .join()
+            .getBody();
+
+    final InventoryEntryDraft inventoryEntryDraft1 =
+        InventoryEntryDraftBuilder.of()
+            .sku("SKU_WITH_CHANNEL")
+            .quantityOnStock(1L)
+            .supplyChannel(ChannelResourceIdentifierBuilder.of().id(channel.getId()).build())
+            .build();
+
+    final InventoryEntryDraft inventoryEntryDraft2 =
+        InventoryEntryDraftBuilder.of()
+            .sku("SKU_WITHOUT_CHANNEL")
+            .quantityOnStock(1L)
+            .supplyChannel(ChannelResourceIdentifierBuilder.of().id(channel.getId()).build())
+            .build();
+
+    CompletableFuture.allOf(
+            inventoryService.createInventoryEntry(inventoryEntryDraft1).toCompletableFuture(),
+            inventoryService.createInventoryEntry(inventoryEntryDraft2).toCompletableFuture())
+        .join();
+
+    // test
+    final Set<InventoryEntryIdentifier> identifiers = new HashSet<>();
+    identifiers.add(InventoryEntryIdentifier.of(inventoryEntryDraft1));
+    identifiers.add(InventoryEntryIdentifier.of(inventoryEntryDraft2));
+
+    final Set<InventoryEntry> result =
+        inventoryService
+            .fetchInventoryEntriesByIdentifiers(identifiers)
+            .toCompletableFuture()
+            .join();
+
+    // assertion
+    assertThat(result).isNotNull();
+    assertThat(result).hasSize(2);
+    // assert references are not expanded
+    result.stream()
+        .filter(inventoryEntry -> inventoryEntry.getSupplyChannel() != null)
+        .map(InventoryEntry::getSupplyChannel)
+        .forEach(supplyChannel -> assertThat(supplyChannel.getObj()).isNull());
+  }
+
+  @Test
+  void createInventoryEntry_ShouldCreateCorrectInventoryEntry() {
+    // prepare draft and create
+    final InventoryEntryDraft inventoryEntryDraft =
+        InventoryEntryDraftBuilder.of()
+            .sku(SKU_2)
+            .quantityOnStock(QUANTITY_ON_STOCK_2)
+            .expectedDelivery(EXPECTED_DELIVERY_2)
+            .restockableInDays(RESTOCKABLE_IN_DAYS_2)
+            .build();
+
+    final Optional<InventoryEntry> result =
+        inventoryService.createInventoryEntry(inventoryEntryDraft).toCompletableFuture().join();
+
+    // assertions
+    assertThat(result)
+        .hasValueSatisfying(
+            inventoryEntry -> {
+              assertThat(inventoryEntry.getQuantityOnStock()).isEqualTo(QUANTITY_ON_STOCK_2);
+              assertThat(inventoryEntry.getRestockableInDays()).isEqualTo(RESTOCKABLE_IN_DAYS_2);
+              assertThat(inventoryEntry.getExpectedDelivery()).isEqualTo(EXPECTED_DELIVERY_2);
+            });
+
+    // assert CTP state
+    final Optional<InventoryEntry> updatedInventoryEntry =
+        getInventoryEntryBySkuAndSupplyChannel(CTP_TARGET_CLIENT, SKU_2, null, null);
+    assertThat(updatedInventoryEntry).isEqualTo(result.get());
```

Comment:
This cannot be equal. One is type Optional and one is type InventoryEntry.

Reply:
[a399ee4](https://github.com/commercetools/commercetools-sync-java/pull/1034/commits/a399ee48648787ac19e6609b5ab925a8c39bdde4)